core/lib/spree/testing_support/factories/user_factory.rb

```diff
@@ -15,8 +15,12 @@
     end
 
     factory :user_with_addresses, aliases: [:user_with_addreses] do
-      ship_address
-      bill_address
+      ship_address do |user_with_addresses|
+        create(:address, user: user_with_addresses.instance)
+      end
+      bill_address do |user_with_addresses|
+        create(:address, user: user_with_addresses.instance)
+      end
```

Comment:
This is necessary because when we create user_with_addresses,
user has foreign keys for ship & bill addresses, but both addresses does not have foreign key for user.